internal/services/page_rule/resource.go

```diff
@@ -110,7 +110,7 @@ func (r *PageRuleResource) Update(ctx context.Context, req resource.UpdateReques
 		return
 	}
 
-	dataBytes, err := data.MarshalJSONForUpdate(*state)
+	dataBytes, err := data.MarshalJSON()
```

Comment:
What are the consequences of using this other method?

Reply:
using `MarshalJSONForUpdate(*state)` gives us incorrect PUT values. 

example: 
```
{
    "actions": {
        "cache_by_device_type": "off"
    },
    "priority": 1,
    "status": "disabled",
    "target": "ppxzakepze.terraform.cfapi.net"
}

```
target / actions are incorrect in this case, it should be
```
{
    "priority": 1,
    "status": "disabled",
    "targets": [
        {
            "constraint": {
                "operator": "matches",
                "value": "mudxsiynjr.terraform.cfapi.net"
            },
            "target": "url"
        }
    ],
    "actions": [
        {
            "id": "cache_by_device_type",
            "value": "off"
        }
    ]
}

```

Reply:
`MarshalJSONForUpdate` should be used in the updates as it takes into considerations the field type (required, optional, computed) to ensure we always have the correct known values. this seems like the fields or schemas may be incorrect defined if it's not picking them.

Reply:
Ah, I see. I wonder if instead we should also give the `PageRuleActionsModel.MarshalJSONForUpdate` method the same treatment we did for `PageRuleActionsModel.MarshalJSON`. This would help keep the amount of code changes to codegen'd files to a minimum.
https://github.com/cloudflare/terraform-provider-cloudflare/blob/ecea695686810d5c7d88001c45f66087ac2712bf/internal/services/page_rule/model.go#L28

Reply:
I'm looking at how the schema is defined here https://github.com/cloudflare/terraform-provider-cloudflare/pull/4722/files#diff-7ff72210193722a4e79ee559520597a59a76770873c73c8040a411320478cef3L62 

and i think Jacob is correct. We probably need `actions` to have  `id` and `value` instead of all the page rules?